connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -160,6 +160,14 @@ private[sql] class AvroDeserializer(
         (logicalDataType, catalystType) match {
           case (LongType, LongType) => (updater, ordinal, value) =>
             updater.setLong(ordinal, value.asInstanceOf[Long])
+          case (_, LongType) => avroType.getLogicalType match {
+            case _: TimestampMicros | _: TimestampMillis |
```

Comment:
Sorry, I didn't get how you can handle in the same way milliseconds and microseconds.

Reply:
good point. @zeruibao what was the behavior before?

Reply:
I always read them as long with `updater.setLong(ordinal, value.asInstanceOf[Long])` In the past, we don't care about LogicalType at all.

Reply:
Could it be a long standing bug?
I am seeing:
https://github.com/apache/spark/pull/41052/files#diff-2eaa6c4b7371cae9800cff1cee843a6245ab91206595cf0e403c077ab2ba80daL138

```
// For backward compatibility, if the Avro type is Long and it is not logical type
        // (the `null` case), the value is processed as timestamp type with millisecond precision.
```

Reply:
I just follow the behavior of this test suite in runtime repo
```
Seq(
    "time-millis",
    "time-micros",
    "timestamp-micros",
    "local-timestamp-millis",
    "local-timestamp-micros"
  ).foreach { timeLogicalType =>
    val timestampSchema = timestampSchemaTemplate.format(timeLogicalType)

    if (DatabricksEdgeConfigs.edgeModeEnabled) {
      test(s"$timeLogicalType type write support of Avro Hive serde") {
        Seq(true, false).foreach { useNativeAvro =>
          withSQLConf(DatabricksSQLConf.CONVERT_METASTORE_AVRO, useNativeAvro) {
            val tableName = "tab1"

            withTable(tableName) {
              // Creates the (non-)partitioned Avro table
              versionSpark.sql(
                s"""
                   |CREATE TABLE $tableName
                   |ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe'
                   |STORED AS
                   |  INPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat'
                   |  OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat'
                   |TBLPROPERTIES ('avro.schema.literal' = '$timestampSchema')
            """.stripMargin
              )

              val insertStmt = s"INSERT OVERWRITE TABLE $tableName SELECT 1000"
              val result = versionSpark.sql(s"SELECT 1000").collect()
              versionSpark.sql(insertStmt)
              assert(versionSpark.table(tableName).collect() === result)
            }
          }
        }
      }
    }
```

Reply:
shall I still follow the original behavior?